zerver/views/streams.py

```diff
@@ -47,7 +47,7 @@ def __init__(self, principal: str) -> None:
 
     @staticmethod
     def msg_format() -> str:
-        return _("User not authorized to execute queries on behalf of '{principal}'")
+        return _("Unable to find user '{principal}'")
```

Comment:
This will also change error message in API call. Confirming with @rishig if that is all right?

Reply:
As per my understanding this error message is raised only when user profile doesn't exist with given principal. are there any other situations under which we give this error message in the API?

Reply:
@abhishek125 when user makes API call, this error message is returned as well as when user change from UI. As mentioned in issue 
```
This error message should not be user visible (it's fine for the API). I'm not sure where else it appears, but for here, we can use "Unable to find user XXX".
```
I would suggest to change the error message in UI only, when request returns error.